plugins/modules/net_tools/snmp_facts.py

```diff
@@ -285,6 +296,9 @@ def main():
         module.fail_json(msg=missing_required_lib('pysnmp'), exception=PYSNMP_IMP_ERR)
 
     cmdGen = cmdgen.CommandGenerator()
+    transport_opts = {'retries': m_args['retries']}
```

Comment:
Is the `retries` option available for all versions of the library that this module supports? If not, better only add it when the value is `> 0`.

Reply:
I don't really know the answer to your question, but I was able to track down that parameter to a 2015 version of the library - btw, the default there is `5`, so I'm wondering if I should match that default here.

Reply:
How about not having a default here, and just pass `retries` if the user explicitly specifies a value? Then the default will be "whatever is the default of the library", and the user can override that.

Reply:
fair enough! coming up